component/loki/source/file/decompresser_test.go

```diff
@@ -0,0 +1,178 @@
+package file
+
+import (
+	"os"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/grafana/loki/clients/pkg/promtail/api"
+	"github.com/grafana/loki/clients/pkg/promtail/client/fake"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/atomic"
+)
+
+type noopClient struct {
+	noopChan chan api.Entry
+	wg       sync.WaitGroup
+	once     sync.Once
+}
+
+func (n *noopClient) Chan() chan<- api.Entry {
+	return n.noopChan
+}
+
+func (n *noopClient) Stop() {
+	n.once.Do(func() { close(n.noopChan) })
+}
+
+func newNoopClient() *noopClient {
+	c := &noopClient{noopChan: make(chan api.Entry)}
+	c.wg.Add(1)
+	go func() {
+		defer c.wg.Done()
+		for range c.noopChan {
+			// noop
+		}
+	}()
+	return c
+}
+
+func BenchmarkReadlines(b *testing.B) {
+	entryHandler := newNoopClient()
+
+	scenarios := []struct {
+		name string
+		file string
+	}{
+		{
+			name: "2000 lines of log .tar.gz compressed",
+			file: "test_fixtures/short-access.tar.gz",
+		},
+		{
+			name: "100000 lines of log .gz compressed",
+			file: "test_fixtures/long-access.gz",
+		},
```

Comment:
nit: Can we put these files in `testdata` instead of `test_fixtures`? `testdata` seems more common in Go projects.